lib/api.js

```diff
@@ -8,9 +8,22 @@
 const Linter = require("./linter");
 
 module.exports = {
-    linter: new Linter(),
     Linter,
     CLIEngine: require("./cli-engine"),
     RuleTester: require("./testers/rule-tester"),
     SourceCode: require("./util/source-code")
 };
+
+let deprecatedLinterInstance = null;
+
+Object.defineProperty(module.exports, "linter", {
+    enumerable: false,
+    configurable: false,
```

Comment:
Why non-configurable?

Reply:
So people won’t override it.

I originally had it print a warning when first accessed, but I [was asked](https://github.com/eslint/eslint/issues/9270#issuecomment-348725286) to remove that feature.

Reply:
Could you elaborate on what this is intended to protect against? I don't see how it would be advantageous for a user to tamper with `linter` in particular -- if someone really wanted to monkeypatch for some reason, they could mutate the `eslint.linter` object or `Linter.prototype` anyway.

I think it would be reasonable for us to try to guard against monkeypatching in general (e.g. by freezing exported objects), but it's not clear to me that only doing this for `linter` has any benefit.

Reply:
Fixed.